gradebookng/tool/src/java/org/sakaiproject/gradebookng/business/GradebookNgBusinessService.java

```diff
@@ -928,13 +929,35 @@ public GradeSaveResponse saveExcuse(final Long assignmentId, final String studen
 		final String storedGrade = this.gradingService.getAssignmentScoreString(gradebook.getUid(), assignmentId,
 				studentUuid);
 
+		// if percentage entry type, reformat the grade, otherwise use points as is
+		String storedGradeAdjusted = storedGrade;
+		final Integer gradingType = gradebook.getGradeType();
+		if (Objects.equals(GradingConstants.GRADE_TYPE_PERCENTAGE, gradingType)) {
+                        // the stored grade represents points so the number needs to be adjusted back to percentage
+                        Double storedGradePoints = new Double("0.0");
+                        if(StringUtils.isNotBlank(storedGrade)){
+                                storedGradePoints = FormatHelper.validateDouble(storedGrade);
+                        }
+
+                        final Double maxPoints = this.getAssignment(assignmentId).getPoints();
+                        final Double storedGradePointsFromPercentage = (storedGradePoints * 100) / maxPoints;
+                        storedGradeAdjusted = FormatHelper.formatDoubleToDecimal(storedGradePointsFromPercentage);
+		}
+		// trim the .0 from the grades if present. UI removes it so lets standardise. Do this twice.
+		storedGradeAdjusted = StringUtils.trimToNull(StringUtils.removeEnd(storedGradeAdjusted, ".0"));
+		storedGradeAdjusted = StringUtils.trimToNull(StringUtils.removeEnd(storedGradeAdjusted, ".0"));
+
+                if (log.isDebugEnabled()) {
+                        log.debug("storedGradeAdjusted: " + storedGradeAdjusted);
+                }
```

Comment:
Using log.isDebugEnabled is old and unnecessary. The correct way is a simple:

log.debug("storedGradeAdjusted: {}", storedGradeAdjusted);

Reply:
We should then correct the same issue in the saveGrade method.

Reply:
@ottenhoff - Should I proceed with fixes to address the comma issue and the archaic log.isDebugEnabled issue?

Reply:
yes please